### PR TITLE
feat: add fieldset-legend component

### PR DESCRIPTION
**Pull request by josokinas**
## Purpose

Add FieldsetLegend component, that is used same as `<legend>` element within `<fieldset>`.

## Approach

Implements FieldsetLegend component.

## Testing

On Storybook preview, including Checkbox and Radio stories "with fieldset legend".

## Risks

N/A

### CODE REVIEW

File: packages/core/src/components/fieldset-legend/fieldset-legend.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+@use '../../helpers';
+
+@mixin FieldsetLegend() {
+  .ods-fieldset-legend {
```
Comment (rabelloo) [inline packages/core/src/components/fieldset-legend/fieldset-legend.scss:4]: How do you feel about `FieldLegend`?

Does it make it seem like it should be a children of `Field`'s instead?

File: packages/react/src/components/checkbox/checkbox.stories.tsx [hunk 2]
```diff
@@ -66,6 +67,34 @@ export const AsIndeterminate = (props: CheckboxProps) => {
...
+  name,
+  legend,
+  ...restCheckboxProps
+}: CheckboxesWithFieldsetLegendProps) => (
+  <Fieldset>
+    <FieldsetLegend>{legend}</FieldsetLegend>
+    <Checkbox {...restCheckboxProps} name={name} value="1">
+      one
+    </Checkbox>
+    <Checkbox {...restCheckboxProps} name={name} value="2">
```
Comment (rabelloo) [inline packages/react/src/components/checkbox/checkbox.stories.tsx:85]: 💅🏻  There's no need to manually destructure the name here.

```suggestion
  legend,
  ...restCheckboxProps
}: CheckboxesWithFieldsetLegendProps) => (
  <Fieldset>
    <FieldsetLegend>{legend}</FieldsetLegend>
    <Checkbox {...restCheckboxProps} value="1">
      one
    </Checkbox>
    <Checkbox {...restCheckboxProps} value="2">
```

File: packages/react/src/components/fieldset-legend/fieldset-legend.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { c, classy } from '@onfido/castor';
+import React from 'react';
+
+/**
+ * Intended to be used within `Fieldset` component, providing a text legend.
```
Comment (rabelloo) [inline packages/react/src/components/fieldset-legend/fieldset-legend.tsx:5]: 💅🏻 

```suggestion
 * Use within `Fieldset` component to provide a legend to an input.
```

File: packages/react/src/components/radio/radio.stories.tsx [hunk 4]
```diff
@@ -57,6 +58,32 @@ export const Disabled = storyOf(Radio, 'disabled', [true, false], {
...
+  name,
+  legend,
+  ...restRadioProps
+}: RadiosWithFieldsetLegendProps) => (
+  <Fieldset>
+    <FieldsetLegend>{legend}</FieldsetLegend>
+    <Radio {...restRadioProps} name={name} value="yes">
+      yes
+    </Radio>
+    <Radio {...restRadioProps} name={name} value="no">
```
Comment (rabelloo) [inline packages/react/src/components/radio/radio.stories.tsx:76]: 💅🏻  same here

```suggestion
  legend,
  ...restRadioProps
}: RadiosWithFieldsetLegendProps) => (
  <Fieldset>
    <FieldsetLegend>{legend}</FieldsetLegend>
    <Radio {...restRadioProps} value="yes">
      yes
    </Radio>
    <Radio {...restRadioProps} value="no">
```

**Review by rabelloo — APPROVED** [review]

File: packages/react/src/components/fieldset-legend/fieldset-legend.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  <Fieldset>
+    <FieldsetLegend {...restFieldsetLegendProps}>
+      {legend}
+      <HelperText>{helperText}</HelperText>
+    </FieldsetLegend>
+  </Fieldset>
+);
+WithHelperText.argTypes = omit<FieldsetLegendWithHelperTextProps>('children');
+WithHelperText.args = {
+  legend: 'Lgend',
```
Comment (rabelloo) [inline packages/react/src/components/fieldset-legend/fieldset-legend.stories.tsx:81]: 💅🏻 

```suggestion
  legend: 'Legend',
```

File: packages/core/src/components/fieldset-legend/fieldset-legend.scss (hunk 1 above)
Comment (josokinas) [reply packages/core/src/components/fieldset-legend/fieldset-legend.scss:4]: Yes, as this is only used with `Fieldset` but should not be used with `Field`.

I think complexity will be if we want to do some functionality on `Field` that is to be also on `Fieldset`, but it's then another thought on how to share and reuse the code.

File: packages/react/src/components/fieldset-legend/fieldset-legend.tsx (hunk 3 above)
Comment (josokinas) [reply packages/react/src/components/fieldset-legend/fieldset-legend.tsx:5]: Will not apply this, as I'd like to avoid using "input" here.

Comment (rabelloo) [reply packages/react/src/components/fieldset-legend/fieldset-legend.tsx:5]: No worries, maybe this then?

```suggestion
 * Use within `Fieldset` component to provide a text legend.
```

**Pull request merged**